### PR TITLE
added download_rate to openapi

### PR DESCRIPTION
**Pull request by juliaElastic**
## What is the problem this PR solves?

// Please do not just reference an issue. Explain WHAT the problem this PR solves here.

Added `download_rate` to openapi spec

## How does this PR solve the problem?

// Explain HOW you solved the problem in your code. It is possible that during PR reviews this changes and then this section should be updated.

## Design Checklist

- [ ] I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.
- [ ] I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.
- [ ] I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc.

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/fleet-server#changelog)

## Related issues

Relates https://github.com/elastic/kibana/issues/171943

### CODE REVIEW

File: model/openapi.yml [hunk 1]
```diff
@@ -283,6 +283,10 @@ components:
           description: The artifact download progress as a percentage.
           type: number
           format: double
+        download_rate:
```
Comment (juliaElastic) [inline model/openapi.yml:286]: download_rate seems optional in agent: https://github.com/elastic/elastic-agent/blob/cc81e4b376068eff7e15330f2c9fa0969d28c8f5/internal/pkg/agent/application/upgrade/details/details.go#L38

**Review by nchaulet — APPROVED** [review]
LGTM!

**Comment by juliaElastic** [comment]
integration test fails with
`Error response from daemon: manifest for docker.elastic.co/elasticsearch/elasticsearch:8.13.0-jqmw70gd-SNAPSHOT-amd64 not found: manifest unknown: manifest unknown`

**Comment by nchaulet** [comment]
Interesting looks like that was merged today https://github.com/elastic/fleet-server/blob/4d155c31683f2afef333f087061fc09b6311805e/dev-tools/integration/.env#L3 and this will not work well with our script that create the docker image

**Comment by juliaElastic** [comment]
@v1v the pinned version was merged, though the [build failed](https://buildkite.com/elastic/fleet-server/builds/2756#018c8143-1942-4eae-ab1b-9936911c03ae). Any idea how to fix this?
It seems there is a missing elasticsearch image:
`Error response from daemon: manifest for docker.elastic.co/elasticsearch/elasticsearch:8.13.0-jqmw70gd-SNAPSHOT-amd64 not found: manifest unknown: manifest unknown`

File: model/openapi.yml (hunk 1 above)
Comment (cmacknz) [inline model/openapi.yml:286]: We also added two more download fields `retry_error_msg` and `retry_until`, let's add those while we are in here:

They came from https://github.com/elastic/elastic-agent/pull/3845

Comment (cmacknz) [reply model/openapi.yml:286]: Relates https://github.com/elastic/kibana/issues/173370

Comment (juliaElastic) [reply model/openapi.yml:286]: Added here: https://github.com/elastic/fleet-server/pull/3177/commits/e0a0188a5f90e979d1f659e28bd64e28cd952002

File: model/openapi.yml [hunk 2]
```diff
@@ -283,6 +283,17 @@ components:
...
+          description: The artifact download rate as bytes per second.
+          type: number
+          format: double 
+        retry_error_msg:
+          description: The error message that is a result of a retryable upgrade.
```
Comment (cmacknz) [inline model/openapi.yml:291]:
```suggestion
          description: The error message that is a result of a retryable upgrade download failure.
```

File: model/openapi.yml [hunk 3]
```diff
@@ -283,6 +283,17 @@ components:
...
+          type: string 
+        retry_until:
+          description: The RFC3339 timestamp the upgrade is retried until.
```
Comment (cmacknz) [inline model/openapi.yml:294]:
```suggestion
          description: The RFC3339 timestamp of the deadline the upgrade download is retried until.
```

**Pull request merged**